main.py

```diff
@@ -2,25 +2,67 @@
 
 import sys
 import logging
-import threading
 from hmi import HMI
 from game import Game
 from PyQt5 import QtWidgets
+import commsettings
+import json
+import argparse
 
+def build_args(loglevel, initial_port, target_scenario):
+    start_port = initial_port
+    HMI_PORT = commsettings.get_port(start_port)
+    GAME_PORT = commsettings.get_port(HMI_PORT)
+    hmi_args = dict()
+    game_args = dict()
 
+    hmi_args['loglevel'] = loglevel
+    game_args['loglevel'] = loglevel
 
-def main(loglevel=logging.INFO):
+    hmi_args['hmi_port'] = HMI_PORT
+    game_args['hmi_port'] = HMI_PORT
+
+    hmi_args['game_port'] = GAME_PORT
+    game_args['game_port'] = GAME_PORT
+
+    hmi_args['ui_file'] = "ui.ui"
+
+    if target_scenario is not None:
+        try:
+            spins = json.loads(target_scenario)['spins']
+        except:
+            spins = None
+        try:
+            players = json.loads(target_scenario)['players']
+        except:
+            players = None
+        try:
+            options = json.loads(target_scenario)['options']
+        except:
+            options = None
+
+        game_args['predetermined_spins'] = spins
+        game_args['predetermined_players'] = players
+        game_args['predetermined_startingplayer'] = options['setStartingPlayer'] if "setStartingPlayer" in options.keys() else None
```

Comment:
this line is throwing `AttributeError: 'NoneType' object has no attribute 'keys'` on me

Reply:
resolved w/ a conditional